src/op_analytics/cli/subcommands/pulls/defillama.py

```diff
@@ -88,60 +83,111 @@ def process_breakdown_stables(data):
         "price",
     ]
 
-    metadata = {}
+    metadata: Dict[str, Optional[str]] = {}
+
+    # Collect required metadata fields
     for key in must_have_metadata_fields:
+        if key not in data:
+            raise KeyError(f"Missing required metadata field: '{key}'")
         metadata[key] = data[key]
 
+    # Collect additional optional metadata fields
     for key in metadata_fields:
         metadata[key] = data.get(key)
 
-    result = pl.DataFrame(rows, infer_schema_length=len(rows)).with_columns(
-        id=pl.lit(metadata["id"]),
-        name=pl.lit(metadata["name"]),
-        symbol=pl.lit(metadata["symbol"]),
+    result = (
+        pl.DataFrame(rows, infer_schema_length=len(rows)).with_columns(
+            id=pl.lit(metadata["id"]),
+            name=pl.lit(metadata["name"]),
+            symbol=pl.lit(metadata["symbol"]),
+        )
+        if rows
+        else pl.DataFrame()
     )
 
     return result, metadata
 
 
-def pull_stables():
-    """Pull stablecoin data from DeFiLlama.
+def pull_stables(
+    stablecoin_ids: Optional[List[str]] = None, days: int = 30
+) -> Dict[str, pl.DataFrame]:
+    """
+    Pulls and processes stablecoin data from DeFiLlama.
+
+    Args:
+        stablecoin_ids: List of stablecoin IDs to process. Defaults to None (process all).
+        days: Number of days of data to retrieve. Defaults to 30.
 
-    - Fetch the stablecoins summary endpoint.
-    - For each stablecoin, fetch the detailed breakdown data.
-    - Write all results to BigQuery.
+    Returns:
+        A dictionary containing metadata and breakdown DataFrames.
     """
     session = new_session()
     summary = get_data(session, SUMMARY_ENDPOINT)
 
+    if "peggedAssets" not in summary:
+        raise KeyError("The 'peggedAssets' key is missing from the summary data.")
+
     urls = {}
     for stablecoin in summary["peggedAssets"]:
         stablecoin_id = stablecoin["id"]
-        urls[stablecoin_id] = BREAKDOWN_ENDPOINT.format(id=stablecoin_id)
+        if stablecoin_ids is None or stablecoin_id in stablecoin_ids:
+            urls[stablecoin_id] = BREAKDOWN_ENDPOINT.format(id=stablecoin_id)
+    if not urls:
+        raise ValueError("No valid stablecoin IDs provided.")
 
-    stablecoin_data = run_concurrently(lambda x: get_data(session, x), urls, max_workers=4)
+    stablecoin_data = run_concurrently(
+        lambda x: get_data(session, x), urls, max_workers=4
+    )
 
-    breakdown_dfs = []
-    metadata_rows = []
-    for data in stablecoin_data.values():
-        breakdown_df, metadata = process_breakdown_stables(data)
-        breakdown_dfs.append(breakdown_df)
-        metadata_rows.append(metadata)
+    breakdown_dfs: List[pl.DataFrame] = []
```

Comment:
nice, love having types! 

Since a recent python version, can't remember which one, you don't need to import the special capitalized `List` or `Dict`.   You can use:
```
breakdown_dfs: list[pl.DataFrame] = []
metadata_rows: list[dict[str, str]] = []
```

Reply:
Yeah it was in the spirit of personal preference, something like "SQL native commands like 'FROM' should be capitalized" to not confuse them with the rest. 

Will revert